plugins/chunter-resources/src/components/EditChannel.svelte

```diff
@@ -0,0 +1,144 @@
+<!--
+//
+// Copyright © 2022 Anticrm Platform Contributors.
+//
+// Licensed under the Eclipse Public License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License. You may
+// obtain a copy of the License at https://www.eclipse.org/legal/epl-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//
+// See the License for the specific language governing permissions and
+// limitations under the License.
+-->
+<script lang="ts">
+  import { Attachments } from '@anticrm/attachment-resources'
+  import { Channel } from '@anticrm/chunter'
+  import type { Class, Ref } from '@anticrm/core'
+  import type { IntlString } from '@anticrm/platform'
+  import { createQuery, getClient, Members } from '@anticrm/presentation'
+  import { EditBox, Icon, IconClose, Label, ActionIcon, Scroller } from '@anticrm/ui'
+  import { createEventDispatcher } from 'svelte'
+
+  import chunter from '../plugin'
+
+  export let _id: Ref<Channel>
+  export let _class: Ref<Class<Channel>>
+
+  let channel: Channel
+
+  const dispatch = createEventDispatcher()
+
+  const client = getClient()
+  const clazz = client.getHierarchy().getClass(_class)
+
+  const query = createQuery()
+  $: query.query(chunter.class.Channel, { _id }, (result) => {
+    channel = result[0]
+  })
+
+  const tabLabels: IntlString[] = [chunter.string.Channel, chunter.string.Members]
+  let selectedTabIndex = 0
+
+  function onNameChange (ev: Event) {
+    const value = (ev.target as HTMLInputElement).value
+    if (value.trim().length > 0) {
+      client.updateDoc(_class, channel.space, channel._id, { name: value })
+    } else {
+      // Just refresh value
+      query.query(chunter.class.Channel, { _id }, (result) => {
+        channel = result[0]
+      })
+    }
+  }
+
+  function onTopicChange (ev: Event) {
+    const newTopic = (ev.target as HTMLInputElement).value
+    client.updateDoc(_class, chunter.space.Topic, channel._id, { topic: newTopic })
```

Comment:
Also you can use `client.update(channel, { topic: newTopic })

Reply:
Done